src/events/types/inventory/InventoryInteractEvent.ts

```diff
@@ -0,0 +1,46 @@
+import { Player } from '../../../player/Player';
+import { Cancellable } from '../../Cancellable';
+import { EventResult } from '../../EventResult';
+import { InventoryEvent } from './InventoryEvent';
+
+/**
+ * An abstract base class for events that describe an interaction between a HumanEntity and the contents of an Inventory.
+ */
+export class InventoryInteractEvent extends InventoryEvent implements Cancellable {
+
+  /**
+   * Gets the player who performed the click.
+   */
+  public getWhoClicked(): Player {
+    return Player.fromJava(this.toJava().getPlayer());
+  }
+
+  /**
+   * Gets the EventResult of this event.
+   */
+  public getResult(): EventResult {
+    return this.toJava().getResult();
+  }
+
+  /**
+   * Sets the result of this event.
+   */
+  public setResult(result: EventResult): void {
+    this.toJava().setResult(result);
+  }
```

Comment:
```ts
const javaResult = Java.resolve('org.bukkit.event.Event.Result').valueOf(result);
this.toJava().setResult(javaResult);
```

Gotta test it out to make sure that works, but I think that's it

Reply:
Same as my comment [here](https://github.com/customrealms/core/pull/7#discussion_r753919838)